apps/api/src/database/repositories/agent-repository.ts

```diff
@@ -19,14 +19,14 @@ import { PartialExcept } from "./types.js";
 export async function create(agent: InsertAgent): Promise<SelectAgent> {
   try {
     const now = new Date();
-    const [result] = await db
-      .insert(agents)
-      .values({
-        ...agent,
-        createdAt: agent.createdAt || now,
-        updatedAt: agent.updatedAt || now,
-      })
-      .returning();
+    const normalizedWalletAddress = agent.walletAddress?.toLowerCase();
```

Comment:
we need to ensure that all inserts or queries that touch `walletAddress` are properly lowercased. else, we'll run into problems with UNIQUE constraints on inserts, or a "no match" on queries when the wallet _does_ exists (e.g., trying to check: checksummed address === all-lowercase address).